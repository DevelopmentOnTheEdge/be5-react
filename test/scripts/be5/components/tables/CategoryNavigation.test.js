import React from 'react';
import renderer from 'react-test-renderer';
import CategoryNavigation from "../../../../../src/scripts/be5/components/tables/CategoryNavigation";


test('snapshot', () => {
  const json = {
    attributes: [{"children":[{"children":[{"children":[],"id":5,"name":"c1"},{"children":[],"id":6,"name":"c2"}],"id":3,"name":"p1"}],"id":2,"name":"Root"}],
    type: "documentCategories"
  };

  let tree = renderer.create(
    <CategoryNavigation data={json} url={"table/publications/Compact view/cat=3"} />
  ).toJSON();
  expect(tree).toMatchSnapshot();

  tree = renderer.create(
      <CategoryNavigation data={json} url={"table/publications/Compact view/cat=2"} />
  ).toJSON();
  expect(tree).toMatchSnapshot();

});
