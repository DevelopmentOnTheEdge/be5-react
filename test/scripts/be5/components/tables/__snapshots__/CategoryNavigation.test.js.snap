// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="category-navigation"
>
  <table>
    <tbody>
      <tr>
        <td>
          <a
            className="d-block"
            href="#!table/publications/Compact%20view/cat=2"
          >
            Root
          </a>
        </td>
        <td>
          <span>
            -&gt;
          </span>
        </td>
        <td>
          <span
            className="d-block"
          >
            p1
          </span>
        </td>
        <td>
          <span>
            -&gt;
          </span>
        </td>
        <td>
          <a
            className="d-block"
            href="#!table/publications/Compact%20view/cat=5"
          >
            c1
          </a>
          <a
            className="d-block"
            href="#!table/publications/Compact%20view/cat=6"
          >
            c2
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`snapshot 2`] = `
<div
  className="category-navigation"
>
  <table>
    <tbody>
      <tr>
        <td>
          <a
            className="d-block"
            href="#!table/publications/Compact%20view"
          >
            Root
          </a>
        </td>
        <td>
          <span>
            -&gt;
          </span>
        </td>
        <td>
          <a
            className="d-block"
            href="#!table/publications/Compact%20view/cat=3"
          >
            p1
          </a>
        </td>
        <td>
          <span>
            -&gt;
          </span>
        </td>
        <td>
          <a
            className="d-block"
            href="#!table/publications/Compact%20view/cat=5"
          >
            c1
          </a>
          <a
            className="d-block"
            href="#!table/publications/Compact%20view/cat=6"
          >
            c2
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
